### ISSUE TITLE
Create dist_map for image segmentation mask as label.

### ISSUE THREAD

**Issue by gr33n1** [LIVIAETS/boundary-loss#48]
Hello,

I put the relevant code in order to preprocess my data.
while loading the data I have preprocess step where I have put your code like the given usage [example](https://github.com/LIVIAETS/boundary-loss/tree/749602d822fcb097c79a7fb708805b8982c37030#usage)
and it works as follows:

```python
    def preprocess(self, sample):
        #my code  - usual preprocess

        if self.transform is not None:
            sample = self.transform(sample)
        image, mask = sample

        #  loss code: at this point, mask is of shape(3,H,W) and values are 0 or 1
        #  each of the channels represent the pixel class(r,g,b)

        disttransform = dist_map_transform([1,1,1], 3) # k=3 for 3 classes
        
        dist_map_tensor = disttransform( mask)

        return image, mask ,dist_map_tensor
```
It is not working well I have many problems, but just as a beginning is this the right configuration for my case?
problems example :
1) in gt_transform ---> class2one_hot returns 5 dim  --- > res.shape  torch.Size([1, 3, 3, 320, 320])
2) after all when computing the loss with dist map the assertion returned from simplex function doesn't hold 
torch.allclose(_sum, _ones)

i need some help to configure

Thanks!

**Comment by HKervadec** [comment]
Hi,

I think the issue is that `mask` (which is already one-hot encoded) goes into the `dist_map_transform` dataloader function, while you could send it directly to `one_hot2dist`. For reference, here is what it does:
```python
def dist_map_transform(resolution: Tuple[float, ...], K: int) -> Callable[[D], Tensor]:
        return transforms.Compose([
                gt_transform(resolution, K),
                lambda t: t.cpu().numpy(),
                partial(one_hot2dist, resolution=resolution),
                lambda nd: torch.tensor(nd, dtype=torch.float32)
        ])


def gt_transform(resolution: Tuple[float, ...], K: int) -> Callable[[D], Tensor]:
        return transforms.Compose([
                lambda img: np.array(img)[...],
                lambda nd: torch.tensor(nd, dtype=torch.int64)[None, ...],  # Add one dimension to simulate batch
                partial(class2one_hot, K=K),
                itemgetter(0)  # Then pop the element to go back to img shape
        ])
```
So, end-to end it gives:
```python
        transforms.Compose([
                lambda img: np.array(img)[...],
                lambda nd: torch.tensor(nd, dtype=torch.int64)[None, ...],  # Add one dimension to simulate batch
                partial(class2one_hot, K=K),
                itemgetter(0)  # Then pop the element to go back to img shape
                lambda t: t.cpu().numpy(),
                partial(one_hot2dist, resolution=resolution),
                lambda nd: torch.tensor(nd, dtype=torch.float32)
        ])
```

What it means is (I believe) you are doing a one-hot-encoding of your one-hot encoding first, before sending to the `one_hot2dist`. This is what I think cause the extra dimensions and breaks some asserts.

The following would probably work (if not, please let me know how):
```python
def preprocess(self, sample):
        #my code  - usual preprocess

        if self.transform is not None:
            sample = self.transform(sample)
        image, mask = sample

        #  loss code: at this point, mask is of shape(3,H,W) and values are 0 or 1
        #  each of the channels represent the pixel class(r,g,b)
        assert one_hot(mask[None, ...])  # I create an extra axis because one_hot is made to check a whole batch
        
        dist_map_tensor = one_hot2dist(mask, resolution=[1, 1, 1])

        return image, mask, dist_map_tensor

```

**Comment by gr33n1** [comment]
Thank you
